app/src/main/java/com/arctouch/codechallenge/di/InjectionMovieDetailsPresenter.java

```diff
@@ -0,0 +1,16 @@
+package com.arctouch.codechallenge.di;
+
+import android.content.Context;
+
+import com.arctouch.codechallenge.movie_details.presentation.MovieDetailsContract;
+import com.arctouch.codechallenge.movie_details.presentation.MovieDetailsPresenter;
+
+import io.reactivex.Scheduler;
+
+public class InjectionMovieDetailsPresenter {
```

Comment:
Have you ever used a Dependency Injection library like Dagger 2? Do you have an opinion on them?

Reply:
Yes, I have already used Dagger 2. I didn’t use in this project because I didn’t feel comfortable because of the time I had and I didn’t practice too much using it. In my opinion it is extremely important for dependency injection, makes things much easier for the developer in a way that it generates all the boilerplate that involves the dependencies of many classes in runtime, so right now it is a traceable code which can be seen in a generated class, let’s say DaggerMoviesListPresenterComponent, which would contain all the dependencies of class MoviesListPresenter. It makes things easier to test as well, because the class I mentioned before for example could be injected on the test and all its dependencies would be set and a ready to use instance could be used.